api/auth/authenticate.js

```diff
@@ -1,34 +1,62 @@
 const logger = require('../logger')('authentication');
 const defaultBcrypt = require('bcryptjs');
+const crypto = require('crypto');
+const jwt = require('jsonwebtoken');
 const defaultUserModel = require('../db').models.user;
 
+// This value doesn't need to be persisted between process instances of the
+// server. It only needs to persist between the request for the nonce and the
+// request to authenticate. Using a random secret on every process restart
+// should somewhat increase security - automatic key cycling!
+const NONCE_SECRET = crypto.randomBytes(64);
+
 module.exports = (
   userModel = defaultUserModel,
   bcrypt = defaultBcrypt
-) => async (username, password, done) => {
+) => async (nonce, password, done) => {
   try {
-    logger.verbose(`got authentication request for [${username}]`);
+    logger.verbose(`got authentication request. decoding jwt...`);
+    let verified = false;
+    try {
+      verified = jwt.verify(nonce, NONCE_SECRET, { algoithms: ['HS256'] });
```

Comment:
typo: `algoithms`

Reply:
Oooh, good catch.  Need to add a test on this one - setting the `algorithms` property when verifying the tokens is strictly necessary, for security reasons.  Otherwise an attacker could set the token header to indicate a `null` algorithm and it'd happily verify it, regardless of the signature.  😬